model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeployment.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2018 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamzi.openshift.dataflow.deployment;
+
+
+import io.streamzi.openshift.dataflow.model.ProcessorConstants;
+import io.streamzi.openshift.dataflow.model.ProcessorFlow;
+import io.streamzi.openshift.dataflow.model.ProcessorLink;
+import io.streamzi.openshift.dataflow.model.ProcessorNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Set of things to deploy to a cloud
+ * @author hhiden
+ */
+public class FlowDeployment {
+    private static final Logger logger = Logger.getLogger(FlowDeployment.class.getName());
+    
+    private String cloudId;
+    private boolean primaryCloud;
+    private List<NodeDeployment> nodeDeployments = new ArrayList<>();
+    private List<LinkDeployment> linkDeployments = new ArrayList<>();
+    
+    public FlowDeployment(String cloudId, boolean primaryCloud, ProcessorFlow flow) {
+        this.cloudId = cloudId;
+        this.primaryCloud = primaryCloud;
+        buildFromFlow(flow);
+    }
+
+    public boolean isPrimaryCloud() {
+        return primaryCloud;
+    }
+    
+    public final void buildFromFlow(ProcessorFlow flow){
+        // Add the nodes that are deployed here
+        flow.getNodes().stream()
+            .filter(node->node.getProcessorType()==ProcessorConstants.ProcessorType.DEPLOYABLE_IMAGE)
+            .filter(node->node.getTargetClouds().containsKey(cloudId))
+            .filter(node->node.getTargetClouds().get(cloudId)>0)
+            .forEach(node->nodeDeployments.add(new NodeDeployment(node, node.getTargetClouds().get(cloudId))));
+        
+
+        // Add the links that are not proxied
+        flow.getLinks().stream()
+            .filter(link->containsNode(link.getSource().getParent()))
+            .filter(link->containsNode(link.getTarget().getParent()))
+            .forEach(link->linkDeployments.add(new LinkDeployment(link, LinkDeployment.LinkType.INTERNAL_LINK, cloudId, cloudId)));
+                
+        // Add Proxies for the links that cannot be left as they are
```

Comment:
not exactly sure I get the logic w/ the entire proxy here :-) 

Is that a local topic, that's "bridging" to something outside ?

Reply:
The idea is that links might be entirely local to a cloud or refer to a remote Kafka. Will Update link to go to a different bootstrap server

Reply:
All links will be the same - just have a bootstrap server and topic name